cmd/zpool/zpool_main.c

```diff
@@ -2471,61 +2474,587 @@ zpool_do_import(int argc, char **argv)
+									NULL},
+	[VS_LATENCY] = {"read", "write", "read", "write", NULL},
+	[VS_QUEUES] = {"read", "write", "read", "write", "scrub", NULL},
+	[VS_HISTO] = {"read", "write", "read", "write", "read", "write", "read",
+			"write", "scrub", NULL},
+};
+
+/*
+ * Return the number of strings in a null-terminated string array.
+ * For example:
+ *
+ * 	const char foo[] = {"bar", "baz", NULL}
+ *
+ * returns 2
+ */
+static uint64_t
+str_array_len(const char *array[])
+{
+	uint64_t count = 0;
+	while (array[count])
+		count++;
+
+	return (count);
+}
+
+/*
+ * Return a default column width for default/latency/queue columns. This does
+ * not include histograms, which have their columns autosized.
+ */
+static unsigned int
+default_column_width(iostat_cbdata_t *cb, enum vdev_stat_flags type)
+{
+	unsigned long column_width = 5; /* Normal niceprint */
+	static unsigned long widths[] = {
+		/*
+		 * Choose some sane default column sizes for printing the
+		 * raw numbers.
+		 */
+		[VS_DEFAULT] = 15, /* 1PB capacity */
+		[VS_LATENCY] = 10, /* 1B ns = 10sec */
+		[VS_QUEUES] = 6,   /* 1M queue entries */
+	};
+
+	if (cb->cb_literal)
+		column_width = widths[type];
+
+	return (column_width);
+}
+
+/*
+ * Utility function to print out a line of dashes like:
+ *
+ * 	--------------------------------  -----  -----  -----  -----  -----
+ *
+ * ...or a dashed named-row line like:
+ *
+ * 	logs                                  -      -      -      -      -
+ *
+ * @cb:				iostat data
+ *
+ * @force_column_width		If non-zero, use the value as the column width.
+ * 				Otherwise use the default column widths.
+ *
+ * @name:			Print a dashed named-row line starting
+ * 				with @name.  Otherwise, print a regular
+ * 				dashed line.
+ */
+static void
+print_iostat_dashes(iostat_cbdata_t *cb, unsigned int force_column_width,
+							const char *name)
+{
+	int i;
+	unsigned int dash_sets;
+	unsigned int namewidth;
+	uint64_t flags = cb->cb_flags;
+	uint64_t f;
+	int idx;
+
+	if (cb->cb_flags & VS_HISTO_M)
+		namewidth = MAX(cb->cb_namewidth, strlen("latency"));
+	else
+		namewidth = cb->cb_namewidth;
+
+	if (name) {
+		namewidth = MAX(cb->cb_namewidth, strlen(name));
+		printf("%-*s", namewidth, name);
+	} else {
+		for (i = 0; i < namewidth; i++)
+			(void) printf("-");
+	}
+
+	/* For each bit in flags */
+	for (f = flags; f; f &= ~BIT(idx)) {
+		unsigned int column_width;
+		idx = ffsll(f) - 1;
+		if (force_column_width)
+			column_width = force_column_width;
+		else
+			column_width = default_column_width(cb, idx);
+
+		/* Figure out how many sets of "-----" we need to print */
+		dash_sets = str_array_len(iostat_bottom_labels[idx]);
+
+		while (dash_sets--) {
+			if (name)
+				printf("  %*s-", column_width - 1, " ");
+			else
+				printf("  %.*s", column_width,
+						"--------------------");
+		}
+	}
+	printf("\n");
+}
+
+
+static void
+__print_iostat_separator(iostat_cbdata_t *cb, unsigned int force_column_width)
```

Comment:
[style] the existing convention for wrapping a function like this is to append `_impl` to the function name.  So in this case it would be `print_iostat_separator_impl()`.  I made this same mistake early on as well which is why you still encounter the __ prefix for this in some places.

That said, I doesn't look like we need to provide a wrapper here.  These functions are all private to `zpool_main.c` so we can update all the callers and don't need to maintain a legacy interface.